pages/api/models/module.ts

```diff
@@ -0,0 +1,47 @@
+import { model, Schema } from "mongoose";
+import { ToolInterface } from "./tool";
+import { SlideInterface } from "./slide";
+
+enum StatusType {
+    COMPLETE = "complete",
+    DRAFT = "draft",
+    PUBLISHED = "published",
+}
+
+export interface ModuleInterface {
+    _id: Schema.Types.ObjectId;
+    title: string;
+    tool: ToolInterface["_id"];
+    slides: [SlideInterface["_id"]];
+    status: StatusType;
+    editing: boolean;
+}
+
+const ModuleSchema = new Schema<ModuleInterface>(
+    {
+        title: {
+            type: String,
+            required: true,
+        },
+        tool: {
+            type: Schema.Types.ObjectId,
```

Comment:
im assuming here that we decided a module doesn't always need to be associated with a tool?